dev/Breadcrumb/Breadcrumb.cpp

```diff
@@ -0,0 +1,611 @@
+    m_itemsRepeaterElementFactory->UserElementFactory(newItemTemplate);
+}
+
+void Breadcrumb::UpdateEllipsisBreadcrumbItemDropdownItemTemplate()
+{
+    const winrt::IInspectable& newItemTemplate = DropdownItemTemplate();
+
+    // Copy the item template to the ellipsis button too
+    if (const auto& ellipsisBreadcrumbItem = m_ellipsisBreadcrumbItem.get())
+    {
+        if (const auto& itemImpl = winrt::get_self<BreadcrumbItem>(ellipsisBreadcrumbItem))
+        {
+            itemImpl->SetFlyoutDataTemplate(newItemTemplate);
+        }
+    }
+}
+
+void Breadcrumb::UpdateBreadcrumbItemsFlowDirection()
+{
+    // Only if some ItemsSource has been defined then we change the BreadcrumbItems flow direction
+    if (ItemsSource())
+    {
+        if (const auto& itemsRepeater = m_itemsRepeater.get())
+        {
+            // Add 1 to account for the leading null
+            int32_t elementCount = m_breadcrumbItemsSourceView.Count() + 1;
+            for (int32_t i{}; i < elementCount; ++i)
+            {
+                const auto& element = itemsRepeater.TryGetElement(i).try_as<winrt::BreadcrumbItem>();
+                element.FlowDirection(FlowDirection());
+            }
+        }
+    }
+}
+
+void Breadcrumb::UpdateItemsRepeaterItemsSource()
+{
+    m_itemsSourceChanged.revoke();
+    m_itemsSourceAsObservableVectorChanged.revoke();
+
+    m_breadcrumbItemsSourceView = winrt::ItemsSourceView(ItemsSource());
+    m_itemsSourceChanged = m_breadcrumbItemsSourceView.CollectionChanged(winrt::auto_revoke, { this, &Breadcrumb::OnRepeaterCollectionChanged });
+
+    if (const auto& itemsRepeater = m_itemsRepeater.get())
+    {
+        if (const auto& itemsSource = ItemsSource())
+        {
+            const auto& incc = [this, itemsSource]() {
+                if (m_breadcrumbItemsSourceView)
+                {
+                    return m_breadcrumbItemsSourceView.try_as<winrt::INotifyCollectionChanged>();
+                }
+                else
+                {
+                    return itemsSource.try_as<winrt::INotifyCollectionChanged>();
+                }
+            }();
+
+            if (incc)
+            {
+                m_itemsSourceAsCollectionChanged = incc.CollectionChanged({ this, &Breadcrumb::OnRepeaterCollectionChanged });
+                m_notifyCollectionChanged.set(incc);
+            }
+            else if (const auto bindableObservableVector = itemsSource.try_as<winrt::IBindableObservableVector>())
+            {
+                m_itemsSourceAsBindableVectorChanged = bindableObservableVector.VectorChanged({ this, &Breadcrumb::OnRepeaterCollectionChanged });
+            }
+            else if (const auto& observableVector = itemsSource.try_as<winrt::IObservableVector<IInspectable>>())
+            {
+                m_itemsSourceAsObservableVectorChanged = observableVector.VectorChanged(winrt::auto_revoke, { this, &Breadcrumb::OnRepeaterCollectionChanged });
+            }
+        }
+    }
```

Comment:
Why is this needed ? Hooking to ItemsSourceViews's Collection changed should suffice no ?

Reply:
Using an ItemsSourceView would be better here. ItemsSourceView covers all these cases for you and is already tested, no need to have the same logic in multiple areas.